test/primitives/array-element-test.js

```diff
@@ -123,6 +123,16 @@ describe('ArrayElement', function() {
       });
     });
 
+    describe('#getIn', function() {
+      var arr = new minim.ArrayElement([
+        {
+          a: [1, 2, 3]
+        }
+      ]);
+
+      expect(arr.getIn([0, 'a', 1]).toValue()).to.equal(2);
```

Comment:
So, just a comment on these tests that they sort of hide the interface. Assuming that elements have equality defined, I would find the following more ilumintating and self documenting:

```javascript

describe('#getIn', function() {
      var arrayElement = new minim.ArrayElement([
        {
          a: [1, 2, 3]
        }
      ]);

      nested_element = new Element(2);
      expect(arrayElement.getIn([0, 'a', 1])).to.equal(nested_element);
    });
```

Calling toValue() all over the place obscures the interface and is less articulate IMO.